From 57f24d88a394d5c937c05f1095da99065cc63662 Mon Sep 17 00:00:00 2001
From: Mauro Rossi <issor.oruam@gmail.com>
Date: Fri, 13 Aug 2021 16:16:24 +0200
Subject: [PATCH] AndroidRuntime: add apex_available to
 libandroid_runtime_threads_headers

Fixes the following building error:

error: frameworks/av/apex/Android.bp:161:1: module "com.android.media.swcodec" variant "android_common_com.android.media.swcodec_image": "libbi
nder" requires "libandroid_runtime_threads_headers" that doesn't list the APEX under 'apex_available'.
...
Consider adding "com.android.media.swcodec" to 'apex_available' property of "libandroid_runtime_threads_headers"

Inspired by similar commit "AndroidRuntime: Expose GetJavaVM"
https://android.googlesource.com/platform/frameworks/base/+/c8a33f1688323596cf29481d22655e2e9679c5d0%5E%21/

Change-Id: Ia541ed30f134ed58b40306b8003bd72300182b2a
---
 apex/media/framework/Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/apex/media/framework/Android.bp b/apex/media/framework/Android.bp
index d963e68d80ec..f9ae942edc55 100644
--- a/apex/media/framework/Android.bp
+++ b/apex/media/framework/Android.bp
@@ -58,6 +58,7 @@ java_library {
     hostdex: true, // for hiddenapi check
     apex_available: [
         "com.android.media",
+        "com.android.media.swcodec",
         "test_com.android.media",
     ],
     min_sdk_version: "29",
-- 
2.25.1

